be/src/storage/data_dir.cpp

```diff
@@ -575,9 +575,9 @@ Status DataDir::update_capacity() {
 
 bool DataDir::capacity_limit_reached(int64_t incoming_data_size) {
     double used_pct = (_disk_capacity_bytes - _available_bytes + incoming_data_size) / (double)_disk_capacity_bytes;
-    int64_t left_bytes = _disk_capacity_bytes - _available_bytes - incoming_data_size;
+    int64_t left_bytes = _available_bytes - incoming_data_size;
 
-    if (used_pct >= config::storage_flood_stage_usage_percent / 100.0 &&
+    if (used_pct >= config::storage_flood_stage_usage_percent / 100.0 ||
```

Comment:
Cannot change this `and` to `or`, this is intended to do so, because we don't want to have a waste of disk space if the disk total capacity is quite large even though the usage percent threshold is reached.

Reply:
I agree with your point of view